### ISSUE TITLE
wish: In gridview show thumbnail and file name of video files

### ISSUE THREAD

**Issue by kollix** [nextcloud/android#490]
### Actual behaviour
In the main view (all files) in grid mode I see thumbnails for images, I see a "note" image and the file name for a sound file, but for video files I see nothing but an "arrow" icon.

### Expected behaviour
Preferred solution: Show any frame image from the video as thumbnail
Alternative solution: show at least also the file name

### Environment data
Android version: 4.4.2

Device model: Samsung Galaxy S4 mini

Stock or customized system: stock

Nextcloud app version: 1.4.0

Nextcloud server version: owncloud 8.2

**Comment by tobiasKaminsky** [comment]
This depends on your server configuration.
You can enable a video preview generator.

**Comment by mario** [comment]
@tobiasKaminsky videos should still have a filename underneath them imho.

**Comment by tobiasKaminsky** [comment]
With our new #567 Photo view we can have both, I think:
- just thumbnails in Photo view
- in FileList if you switch to grid view show FileName for videos

cc @eppfel @jancborchardt

**Comment by eppfel** [comment]
mh, videos are not as easy to discover than images, because they have 3 dimensions and a thumb only 2. So i think it is a good idea to add the video filename. But I don't know of a visually appealing solution for that. 

A gif preview would be awesome 😍

**Comment by jancborchardt** [comment]
Yeah … videos should ideally not have a name either, it's mostly as useless as the names of photos.

Only a "video" icon on them (check how Instagram does it, you can use the video.svg icon from the server), and ideally an animated preview of 3 or 5 frames of different sections of the video.

**Comment by tobiasKaminsky** [comment]
> Only a "video" icon on them (check how Instagram does it, you can use the video.svg icon from the server)

We have this already. If the server supports video preview we have an image with a "play" icon on it.

**Comment by jancborchardt** [comment]
So on Nextcloud 11+ (or which version) this issue does not occur? Cause I just saw the issue was reported with 8.2 which is pretty outdated.

**Comment by tobiasKaminsky** [comment]
It depends if the server supports video previews as we rely on their preview API.

**Comment by jancborchardt** [comment]
So maybe enabling previews for video files is something for the "Tips & tricks" section of the server? cc @rullzer @icewind1991

**Comment by tobiasKaminsky** [comment]
So what to do now?

**Comment by jancborchardt** [comment]
If we detect the server does not support video previews, show the video filetype icon as placeholder, instead of the play icon - and also the filename underneath. But no filename if the video preview image can be shown. :)
